swagger_server/itm/itm_action_handler.py

```diff
@@ -0,0 +1,416 @@
+                    time_passed = self.times_dict["treatmentTimes"][supply_used]
+                break
+
+        # Injuries and certain basic vitals are discovered when a casualty is treated.
+        for isso_casualty in self.current_isso.scenario.state.casualties:
+            if isso_casualty.id == casualty.id:
+                casualty.vitals.breathing = isso_casualty.vitals.breathing
+                casualty.vitals.conscious = isso_casualty.vitals.conscious
+                casualty.vitals.mental_status = isso_casualty.vitals.mental_status
+                self._reveal_injuries(isso_casualty, casualty)
+                casualty.visited = True
+
+        # Finally, log the action and return the elapsed time
+        self.session.history.add_history(
+            "Apply Treatment", {"Session ID": self.session.session_id,
+                                "Casualty ID": casualty.id, "Parameters": action.parameters},
+            self.session.scenario.state.to_dict())
+        return time_passed
+
+
+    def check_all_vitals(self, casualty: Casualty):
+        """
+        Check all vitals of the specified casualty in the scenario.
+
+        Args:
+            casualty: The casualty to check.
+        """
+        for isso_casualty in self.current_isso.scenario.state.casualties:
+            if isso_casualty.id == casualty.id:
+                casualty.vitals = isso_casualty.vitals
+                self._reveal_injuries(isso_casualty, casualty)
+                casualty.visited = True
+                self.session.history.add_history(
+                    "Check All Vitals",
+                    {"Session ID": self.session.session_id, "Casualty ID": casualty.id},
+                    casualty.vitals.to_dict())
+                return self.times_dict['CHECK_ALL_VITALS']
+
+
+    def check_pulse(self, casualty: Casualty):
+        """
+        Process checking the pulse (heart rate) of the specified casualty.
+
+        Args:
+            casualty: The casualty to check.
+        """
+        for isso_casualty in self.current_isso.scenario.state.casualties:
+            if isso_casualty.id == casualty.id:
+                casualty.vitals.breathing = isso_casualty.vitals.breathing
+                casualty.vitals.conscious = isso_casualty.vitals.conscious
+                casualty.vitals.mental_status = isso_casualty.vitals.mental_status
+                casualty.vitals.hrpmin = isso_casualty.vitals.hrpmin
+                self._reveal_injuries(isso_casualty, casualty)
+                casualty.visited = True
+                self.session.history.add_history(
+                    "Check Pulse",
+                    {"Session ID": self.session.session_id, "Casualty ID": casualty.id},
+                    casualty.vitals.hrpmin)
+                return self.times_dict['CHECK_PULSE']
+
+
+    def check_respiration(self, casualty: Casualty):
+        """
+        Check the respiration of the specified casualty in the scenario.
```

Comment:
check_pulse and check_respiration are nearly identical and share a lot of code with check_all_vitals.  If we add any other checks, we might consider refactoring these to pull out all the common code or have a generic check_vitals that takes an additional array of things to check (e.g., ['PULSE', 'RESPIRATION']) and sets any item in the array in addition to the basic checks of breathing, consciousness, and mental status.

Reply:
Agreed.  We'll likely be adding blood ox level, so that change is probably in our future.